### PR TITLE
feat(aws): Add private certificate authorities

### PR DESCRIPTION
**Pull request by adam-tylr**
#### Summary

Adds support for acmpca certificate authorities.

### CODE REVIEW

File: plugins/source/aws/resources/services/acmpca/certificate_authorities.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	"github.com/cloudquery/plugin-sdk/v2/transformers"
+)
+
+func CertificateAuthorities() *schema.Table {
+	tableName := "aws_acmpca_certificate_authorities"
+	return &schema.Table{
+		Name:        tableName,
+		Description: `https://docs.aws.amazon.com/privateca/latest/APIReference/API_CertificateAuthority.html`,
+		Resolver:    fetchAcmpcaCertificateAuthorities,
+		Multiplex:   client.ServiceAccountRegionMultiplexer(tableName, "acmpca"),
```
Comment (bbernays) [inline plugins/source/aws/resources/services/acmpca/certificate_authorities.go:19]: The name of the service in this location has to match the name that AWS [uses here](https://github.com/cloudquery/cloudquery/blob/66eefda579565b4e76647daf36ca21070d338f59/plugins/source/aws/client/data/partition_service_region.json#L79)

so in this case the service name is `acm-pca`

File: plugins/source/aws/resources/services/acmpca/certificate_authorities.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		if err != nil {
+			return err
+		}
+		res <- output.CertificateAuthorities
+	}
+	return nil
+}
+
+func resolveCertificateAuthorityTags(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	certAuthority := resource.Item.(*types.CertificateAuthority)
```
Comment (bbernays) [inline plugins/source/aws/resources/services/acmpca/certificate_authorities.go:59]:
```suggestion
	certAuthority := resource.Item.(types.CertificateAuthority)
```

File: plugins/source/aws/resources/services/acmpca/certificate_authorities.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+			},
+		},
+	}
+}
+
+func fetchAcmpcaCertificateAuthorities(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
+	cl := meta.(*client.Client)
+	svc := cl.Services().Acmpca
+	var input acmpca.ListCertificateAuthoritiesInput
+	paginator := acmpca.NewListCertificateAuthoritiesPaginator(svc, &input)
```
Comment (bbernays) [inline plugins/source/aws/resources/services/acmpca/certificate_authorities.go:45]:
```suggestion
	paginator := acmpca.NewListCertificateAuthoritiesPaginator(svc, nil)
```

**Review by bbernays — CHANGES_REQUESTED** [review]
Overall this looks great! A few minor tweaks to get the tests passing!

**Comment by adam-tylr** [comment]
Thanks @bbernays, `make test` is passing now

**Comment by bbernays** [comment]
@adam-tylr Can you run `go mod tidy`?

**Review by bbernays — APPROVED** [review]
LGTM! Once the tests all pass this should merge

**Comment by adam-tylr** [comment]
@bbernays Any idea about the failed check?

**Comment by bbernays** [comment]
@adam-tylr - That test is now passing 😄 ... But unfortunately there are conflicts in the go mod now 😭 ...

**Review by hermanschaaf — APPROVED** [review]

**Comment by hermanschaaf** [comment]
Thanks for this contribution @adam-tylr, looks great!

**Pull request merged**